eth/catalyst/simulated_beacon.go

```diff
@@ -181,7 +192,7 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal, timestamp u
 	// behavior, the pool will be explicitly blocked on its reset before
 	// continuing to the block production below.
 	if err := c.eth.APIBackend.TxPool().Sync(); err != nil {
-		return fmt.Errorf("failed to sync txpool: %w", err)
+		return &errTxPoolTerminated{fmt.Errorf("failed to sync txpool: %w", err)}
```

Comment:
I would just turn it into, not really a need for wrapping another error `type errTxPoolTerminated error`

Reply:
I fear that I have not clearly undrestood this suggestion. But, I did something to move the "failed to sync txpool:" part of the message into the `errTxPoolTerminated.Error` implementation.